stage0/enter.go

```diff
@@ -27,21 +27,28 @@ import (
 )
 
 // Enter enters the container by exec()ing the stage1's /enter similar to /init
-// /enter can expect to have its CWD set to the container root
+// /enter can expect to have its CWD set to the container root.
+// enterPath optionally specifies the path of the enter binary; if empty, it is
+// assumed to be in the stage1 rootfs.
 // imageID and command are supplied to /enter on argv followed by any arguments
-func Enter(cdir string, imageID *types.Hash, cmdline []string) error {
+func Enter(cdir string, imageID *types.Hash, enterPath string, cmdline []string) error {
 	if err := os.Chdir(cdir); err != nil {
 		return fmt.Errorf("error changing to dir: %v", err)
 	}
 
 	id := types.ShortHash(imageID.String())
 
-	ep, err := getStage1Entrypoint(cdir, enterEntrypoint)
-	if err != nil {
-		return fmt.Errorf("error determining entrypoint: %v", err)
-	}
+	var argv []string
+	if enterPath == "" {
+		ep, err := getStage1Entrypoint(cdir, enterEntrypoint)
```

Comment:
This branch could go away if the change I suggested in ``runEnter()`` actually works.